hand-written.md

```diff
@@ -1,39 +1,32 @@
 
-As usual for milestones, 2.12.0-M3 is not binary compatible with any other Scala release, including other 2.12 milestones.
+We'd especially like to thank Felix Mulder for his [excellent work on the new Scaladoc interface](https://github.com/scala/scala/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Amerged+author%3Afelixmulder++milestone%3A2.12.0-M4)! [Check it out!](http://www.scala-lang.org/files/archive/api/2.12.0-M4/)
 
-## About Scala 2.12
 
-The Scala 2.12 series targets Java 8. Programs written in Scala 2.12 must be compiled and run on Java 8 or newer.
+Except for the breaking changes listed below, code that compiles on 2.11.x without deprecation warnings should compile on 2.12.x too, unless you use experimental APIs such as reflection.  If you find incompatibilities, please [file an issue](https://issues.scala-lang.org).
 
-### Source compatibility
+As usual for milestones, 2.12.0-M4 is not binary compatible with any other Scala release, including other 2.12 milestones. Scala 2.12 requires a Java 8 runtime.
 
-Scala 2.12 is mostly source compatible with 2.11.  Code that compiles on 2.11.x without deprecation warnings should compile on 2.12.x too, unless you use experimental APIs such as reflection.  If you find incompatibilities, please [file an issue](https://issues.scala-lang.org).
+### New features
 
-### Binary compatibility
+With M4, we consider 2.12.x feature complete. Please try it out while we still have time to fix any regressions in M5! For the next milestone, we'll focus on addressing any issues discovered with M4, overall robustness and polish, and some smaller improvements that are less likely to cause regressions or source incompatibilities. For RC1, we will try to remain binary compatible with M5, and we won't risk regressions except for the most critical bugs.
 
-Since Scala 2.11, minor releases of Scala are binary compatible with each other.
-Scala 2.12 will continue this tradition: every 2.12.x release will be binary compatible with 2.12.0.
-Milestone releases and release candidates, however, are **not** binary compatible with any other release.
+#### Trait compiles to an interface
+With Java 8 allowing concrete methods in interfaces, Scala 2.12 is able to compile a trait to a single interface. Before, a trait was represented as a class that held the method implementations and an interface. Note that the compiler still has quite a bit of magic to perform behind the scenes, so that care must be taken if a trait is meant to be implemented in Java. (Briefly, the following features require synthetic code in real classes that extend traits that do any of the following: defining fields, calling super, trait body statements, extending a class, relying on linearization to find implementations in the right super trait.)
```

Comment:
instead of:

> Briefly, the following features require synthetic code in real classes that extend traits that do any of the following

maybe:

> Briefly, using any of the following trait features causes the compiler to add synthetic code to any class which extends that trait: ...